src/plugins/embeddable/public/lib/containers/container.ts

```diff
@@ -39,19 +46,29 @@ export abstract class Container<
     [key: string]: IEmbeddable<any, any> | ErrorEmbeddable;
   } = {};
 
-  private subscription: Subscription;
+  private subscription: Subscription | undefined;
 
   constructor(
     input: TContainerInput,
     output: TContainerOutput,
     protected readonly getFactory: EmbeddableStart['getEmbeddableFactory'],
-    parent?: Container
+    parent?: IContainer,
+    initializeSettings?: EmbeddableContainerInitializeSettings
   ) {
     super(input, output, parent);
     this.getFactory = getFactory; // Currently required for using in storybook due to https://github.com/storybookjs/storybook/issues/13834
+
+    // initialize all children on the first input change. Debounced by 1ms so it is run after the constructor is finished.
+    this.getInput$()
+      .pipe(debounceTime(1), first())
```

Comment:
nit: maybe `delay(0)` can be used instead of `debounceTime(1)` as it should be more efficient, and maybe `take(1)` instead of `first()`, as `first()` will throw an error if no value was emitted by `this.getInput$()`.

```suggestion
      .pipe(delay(0), take(1))
```

Reply:
Good call! I've replaced these.